java/src/main/java/com/genexus/cache/redis/RedisClient.java

```diff
@@ -26,105 +27,96 @@
 
 
-public class RedisClient implements ICacheService2, Closeable{
+public class RedisClient implements ICacheService2, Closeable {
 	public static final ILogger logger = LogManager.getLogger(RedisClient.class);
 	private String keyPattern = "%s_%s_%s"; //Namespace_KEY
-	private static int UNDEFINED_PORT = -1;
 	private static int REDIS_DEFAULT_PORT = 6379;
 	private JedisPool pool;
-	private ObjectMapper objMapper; 	
-	public RedisClient() throws IOException {
+	private ObjectMapper objMapper;
+
+	public RedisClient() throws URISyntaxException {
 		initCache();
 	}
 
-	private void initCache() throws IOException {
-		objMapper = new ObjectMapper();
-		objMapper.setVisibility(PropertyAccessor.FIELD, Visibility.ANY);
-		objMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
-		objMapper.enable(SerializationFeature.INDENT_OUTPUT);
+	public RedisClient(String hostOrRedisURL, String password, String cacheKeyPattern) throws URISyntaxException {
+		initCache(hostOrRedisURL, password, cacheKeyPattern);
+	}
 
+	public JedisPool getConnection() {
+		return pool;
+	}
+
+	private void initCache() throws URISyntaxException {
 		GXService providerService = Application.getGXServices().get(GXServices.CACHE_SERVICE);
 		String addresses = providerService.getProperties().get("CACHE_PROVIDER_ADDRESS");
 		String cacheKeyPattern = providerService.getProperties().get("CACHE_PROVIDER_KEYPATTERN");
 		String password = providerService.getProperties().get("CACHE_PROVIDER_PASSWORD");
+		initCache(addresses, password, cacheKeyPattern);
+	}
 
-		if (!isNullOrEmpty(cacheKeyPattern))
-			keyPattern = cacheKeyPattern;
-
-		if (!isNullOrEmpty(addresses)){
+	private void initCache(String hostOrRedisURL, String password, String cacheKeyPattern) throws URISyntaxException {
+		keyPattern = isNullOrEmpty(cacheKeyPattern) ? keyPattern : cacheKeyPattern;
+		String host = "127.0.0.1";
+		hostOrRedisURL = isNullOrEmpty(hostOrRedisURL) ? host: hostOrRedisURL;
+		int port = REDIS_DEFAULT_PORT;
 
-			if (!isNullOrEmpty(password)) {
+		boolean isRedisURIScheme = hostOrRedisURL.startsWith("redis://");
+		String sRedisURI = isRedisURIScheme ? hostOrRedisURL : "redis://" + hostOrRedisURL;
 
-				addresses = "redis://:" + password.trim() + "@" + addresses.trim();
-				try {
-					URI redisUri = new URI(addresses);
-					if (redisUri.getPort()==UNDEFINED_PORT){
-						redisUri = new URI(addresses + ":" + REDIS_DEFAULT_PORT);
-					}
-					pool = new JedisPool(new JedisPoolConfig(), redisUri);
-				}catch (java.net.URISyntaxException ex){
-					logger.error("Invalid redis uri " + addresses, ex);
-				}
+		try {
+			URI redisURI = new URI(sRedisURI);
+			host = redisURI.getHost();
+			if (redisURI.getPort() > 0) {
+				port = redisURI.getPort();
 			}
-		}else{
-			addresses ="127.0.0.1:" + REDIS_DEFAULT_PORT;
+		} catch (URISyntaxException e) {
+			logger.error(String.format("Could not parse Redis URL. Check for supported URLs: %s" + sRedisURI), e);
```

Comment:
It seems like format method dont have an argument.
May be you want to write this way
logger.error(String.format("Could not parse Redis URL. Check for supported URLs: %s", sRedisURI), e);

Reply:
Ooops, fixed!